### DISCUSSION TITLE
Detected Possible Cross-site Scripting on https://example.com/_next/?%3cscRipt%3enetsparker(0x0000BF)%3c%2fscRipt%3e  types urls

### DISCUSSION THREAD

**Discussion by sadam1807** [vercel/next.js#44586] [Help] [answered]
After running Pen test on website with the help of netsparker tool, found  Cross-site Scripting vulnerability on /_next/ URLs,  how we can resolve it ?

 .
![image](https://user-images.githubusercontent.com/42937204/210739667-e51361f5-dddf-4d78-bbcb-44e8131e1d6d.png)

**Comment by icyJoseph** [comment]
Why does your site return redirect? Are you doing any special catch all routes, or have `rewrites`/`redirect` rules that might be too broad? I tested a couple of sites of mine and they straight up return 404 for that given input. However, when I run it on `vercel.com` I do see the redirect behavior.

What Next.js version do you have? I want to try and see if this is out of the box in some version.

**Comment by sadam1807** [comment]
Are you doing any special catch all routes, or have rewrites/redirect rules that might be too broad? **No**

What Next.js version do you have? **Next 11.1.4**

**Reply by icyJoseph** [reply]
Alright, but could you describe a bit more of your routing? Can you make a minimal repro demo?

**Reply by icyJoseph** [reply]
Just to expand further, a create-next-app, out of the box doesn't do the redirect, so it must be something one walks into. And of course, one can also walk out of it.

**Comment by sadam1807** [comment, marked as answer]
Upgrading next js from 11 to 12 resolved this issue.

Thank you.
